suprb/optimizer/rule/ns/archive.py

```diff
@@ -0,0 +1,53 @@
+from abc import ABCMeta, abstractmethod
+
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+from suprb.utils import RandomState
+
+
+class Archive(BaseComponent, metaclass=ABCMeta):
+    """Base Archive Class to store the rules from previous populations"""
+
+    def __init__(self):
+        self.archive = []
+
+    def set_archive(self, pool: list[Rule]):
+        self.archive = pool
+
+    def set_random_state(self, random_state: RandomState):
+        self.random_state_ = random_state
+
+    def get_archive(self):
+        return self.archive
+
+    def extend_archive(self, rules: list[Rule], n: int):
```

Comment:
add some documentation to explain where this differs from set_archive.
I think we could move this to `__call__` to keep it more in line with the rest of the code